File: EC.APP/src/app/models/tenant.ts
export interface ITenant {
    tenantId: number;
    id?: string;
    name: string;
    domain: string;
    plans: 'Starter' | 'Pro' | 'Enterprise';
    status: 'Active' | 'Inactive';
    users: number;

    maxUsers?: number;
    storageLimitGB?: number;
    subscriptionStart?: string;
    subscriptionEnd?: string;

    row?: number;
    totalRowCount?: number;
    flag?: number;
}

File: EC.API/Models/Tenant.cs
namespace EC.API.Models;

public class Tenant
{
    public int TenantId { get; set; }
    public string Id { get; set; }        // external GUID if needed
    public string Name { get; set; }
    public string Domain { get; set; }
    public string Plans { get; set; }      // Starter | Pro | Enterprise
    public string Status { get; set; }    // Active | Inactive
    public int Users { get; set; }
    public int IsDefault { get; set; }
    public int Flag { get; set; }
    public int TotalRowCount { get; set; }
}
